src/js/main.js

```diff
@@ -112,6 +113,144 @@ class HomeTimeline extends React.Component {
 	}
 }
 
+class ReplyTweetModal extends React.Component {
+	constructor (props) {
+	    super(props);
+	    this.state = {
+	      value: '',
+	      replyTweetMessage: null,
+	      isReplyTweetError: null
+	    };
+	    this.handleChange = this.handleChange.bind(this);
+	    this.handleKeyPress = this.handleKeyPress.bind(this);
+	    this.handleReplyTweet = this.handleReplyTweet.bind(this);
+	    this.replyTweetResultOutput = this.replyTweetResultOutput.bind(this);
+	    this.replyTweetResultClass = this.replyTweetResultClass.bind(this);
+	    this.renderingInReplyToTweet = this.renderingInReplyToTweet.bind(this);
+	}
+
+	handleReplyTweet(tweet, replyId) {
+		replyTweet(tweet, replyId).then(response => response.json())
+						  .then(data => {
+						  	this.setState ({
+						  		replyTweetMessage: data.message,
+						  		isReplyTweetError: false
+						  	})
+						  })
+						  .catch(error => {
+						  	this.setState ({
+						  		replyTweetMessage: null,
+						  		isReplyTweetError: true
+						  	})
+						  })
+	}
+
+	handleChange(event) {
+		this.setState({
+			value: event.target.value
+		})
+	}
+
+	renderingInReplyToTweet(tweetObj) {
+		return renderTweetObj(tweetObj, 0, true);
+	}
+
+	handleKeyPress(event) {
+		if(event.key == 'Enter' && this.state.value != ' ') {
```

Comment:
you shouldn't have a key event for enter on the textarea since text areas allow newlines